resources/istio/templates/job-migration-remove-dr.yaml

```diff
@@ -0,0 +1,27 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    helm.sh/hook: "pre-upgrade"
+    helm.sh/hook-weight: "5"
+  name: remove-dr
+spec:
+  activeDeadlineSeconds: 1000
+  backoffLimit: 3
+  template:
+    spec:
+      containers:
+        -
+          command:
+            - bash
+            - "-c"
+            - |
+              set -e
+              kubectl delete dr istio-ingressgateway -n istio-system
```

Comment:
nit :  we can prepare ourselves for the future :D by addidn --ignore-not-found  flag. That will prevent upgrade plan from failing after 1.12 before we remove this delete